src/generators/cfml.ts

```diff
@@ -0,0 +1,114 @@
+import * as util from "../util.js";
+import type { Request } from "../util.js";
+
+import jsesc from "jsesc";
+
+const quote = (str: string): string => {
+  return jsesc(str, { quotes: "single" }).replace(/"/g, '""');
+};
+
+export const _toCFML = (request: Request): string => {
+  let cfmlCode = "";
+
+  cfmlCode += "httpService = new http();\n";
+  cfmlCode += 'httpService.setUrl("' + quote(request.url as string) + '");\n';
+  cfmlCode += 'httpService.setMethod("' + quote(request.method) + '");\n';
+
+  if (request.cookies) {
+    for (const [headerName, headerValue] of request.cookies) {
+      cfmlCode +=
+        'httpService.addParam(type="cookie", name="' +
+        quote(headerName) +
+        '", value="' +
+        quote(headerValue) +
+        '");\n';
+    }
+    util.deleteHeader(request, "Cookie");
+  }
+
+  if (request.headers && request.headers.length) {
+    for (const [headerName, headerValue] of request.headers) {
+      cfmlCode +=
+        'httpService.addParam(type="header", name="' +
+        quote(headerName) +
+        '", value="' +
+        quote(headerValue as string) +
+        '");\n';
+    }
+  }
+
+  if (request.timeout) {
+    cfmlCode +=
+      'httpService.setTimeout(' + (parseInt(request.timeout) || 0) + ');\n';
+  }
+
+  if (request.proxy) {
+    const proxyPart = (request.proxy as string).match(/:([0-9]{2,})/);
+    const proxyPort = proxyPart && proxyPart.length > 1 ? proxyPart[1] : "";
+    const proxy = request.proxy.replace(":" + proxyPort, "");
```

Comment:
This will remove `:` in the URL's "`://`" when there's no port and also we might as well handle a single digit port number.

Curl will choose 1080 as the default port if one isn't provided https://curl.se/docs/manpage.html#-x
Does CFML do that too? 

```suggestion
    let proxy = request.proxy
    let proxyPort = "";
    const proxyPart = (request.proxy as string).match(/:([0-9]+)/);
    if (proxyPart) {
      proxy = request.proxy.slice(0, proxyPort.index);
      proxyPort = proxyPart[1]
    }
```

Reply:
It looks like CFML will default to 80 so maybe we need to set it to 1080 since it is expected to match curl defaults